pkg/cmd/cli/cmd/create_route.go

```diff
@@ -0,0 +1,305 @@
+// and will return the route name and port.
+func resolveRouteInfo(f *clientcmd.Factory, cmd *cobra.Command) (name, intstr.IntOrString, error) {
+	_, kc, err := f.Clients()
+	if err != nil {
+		return "", intstr.IntOrString{}, err
+	}
+
+	ns, err := f.DefaultNamespace()
+	if err != nil {
+		return "", intstr.IntOrString{}, err
+	}
+
+	serviceName := cmdutil.GetFlagString(cmd, "expose")
+	svc, err := kc.Services(ns).Get(serviceName)
+	if err != nil {
+		return "", intstr.IntOrString{}, err
+	}
+
+	name, err := kcmd.NameFromCommandArgs(cmd, args)
+	if err != nil {
+		name = svc.Name
+	}
+
+	var (
+		routePort   intstr.IntOrString
+		supportsTCP bool
+	)
+	for _, port := range svc.Spec.Ports {
+		if port.Protocol == kapi.ProtocolTCP {
+			portString := cmdutil.GetFlagString(cmd, "port")
+			if len(port.Name) > 0 && len(portString) == 0 {
+				// If the service has multiple ports and the user didn't specify --port,
+				// then default the route port to a service port name.
+				routePort = intstr.FromString(port.Name)
+			} else if len(portString) > 0 {
+				// The user specified --port.
+				i, err := strconv.Atoi(portString)
+				if err != nil {
+					routePort = intstr.FromString(portString)
+				} else {
+					routePort = intstr.FromInt(i)
+				}
+			}
+			supportsTCP = true
+			break
+		}
+	}
+	if !supportsTCP {
+		return "", intstr.IntOrString{}, fmt.Errorf("service/%s doesn't support TCP", serviceName)
+	}
+	return name, routePort, nil
+}
+
+const (
+	edgeRouteLong = `
+Create a route that uses edge TLS termination.
+
+Specify the service that the new route should expose via the --expose flag.`
+
+	edgeRouteExample = `  # Create a new route named "my-route" that exposes frontend service.
+  $ %[1]s create route edge my-route --expose frontend
+
+  # Create a new edge route and specify a path. If the route name is omitted,
+  # the service name will be re-used.
+  $ %[1]s create route edge --expose frontend --path /assets`
+)
+
+// NewCmdCreateEdgeRoute is a macro command to create a new route.
+func NewCmdCreateEdgeRoute(f *clientcmd.Factory, cmdOut io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "edge [NAME] [options]",
+		Short:   "Create a route that uses edge TLS termination",
+		Long:    edgeRouteLong,
+		Example: edgeRouteExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := CreateEdgeRoute(f, cmdOut, cmd, args)
+			cmdutil.CheckErr(err)
+		},
+	}
+
+	cmd.Flags().String("path", "", "Path that the router watches to route traffic to the service.")
+	cmd.Flags().String("cert-file", "", "Provide certificate contents.")
+	cmd.Flags().String("key-file", "", "Provide key contents.")
+	cmd.Flags().String("ca-cert-file", "", "Provide cert authority certificate contents.")
+
+	return cmd
+}
+
+// CreateEdgeRoute implements the behavior to run the create edge route command.
+func CreateEdgeRoute(f *clientcmd.Factory, cmdOut io.Writer, cmd *cobra.Command, args []string) error {
```

Comment:
I would have expected a helper to get the certs in the correct format and a straight assignments, so something more like:

```go
route := routeapi.Route{}
route.Name = args[0]
route.TLS.Termination = edge
route.TLS.CertFile = readCert(cert flag)

routeClient.Namespace(ns).Create(route)

```